src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -63,16 +63,47 @@ Copyright (c) .NET Foundation. All rights reserved.
     </BuildOutputInPackage>
   </ItemDefinitionGroup>
 
-  <Target Name="_GetOutputItemsFromPack"
-          Returns="@(_OutputPackItems)">
+  <PropertyGroup>
+    <RestoreOutputPath Condition="'$(RestoreOutputPath)' == ''">$(MSBuildProjectExtensionsPath)</RestoreOutputPath>
+    <PackageOutputPath Condition="'$(PackageOutputPath)' == ''">$(BaseOutputPath)</PackageOutputPath>
+    <NuspecOutputPath Condition="'$(NuspecOutputPath)' == ''">$(BaseIntermediateOutputPath)</NuspecOutputPath>
+  </PropertyGroup>
 
-    <ConvertToAbsolutePath Paths="$(PackageOutputPath)">
-      <Output TaskParameter="AbsolutePaths" PropertyName="PackageOutputAbsolutePath" />
-    </ConvertToAbsolutePath>
+  <!--
+    ============================================================
+    _GetAbsoluteOutputPathsForPack
+    Gets the Absolute output paths for Pack.
+    ============================================================
+  -->
+  <Target Name="_GetAbsoluteOutputPathsForPack">
+    <PropertyGroup>
+      <RestoreOutputPath Condition="'$(RestoreOutputPath)' == ''">$(MSBuildProjectExtensionsPath)</RestoreOutputPath>
+      <PackageOutputPath Condition="'$(PackageOutputPath)' == ''">$(BaseOutputPath)</PackageOutputPath>
+      <NuspecOutputPath Condition="'$(NuspecOutputPath)' == ''">$(BaseIntermediateOutputPath)</NuspecOutputPath>
+    </PropertyGroup>
 
+    <ConvertToAbsolutePath Paths="$(RestoreOutputPath)">
```

Comment:
Looks like the changeset has gotten very large right now. 
I would avoid refactoring too many things.
Please remove the white space changes.

Reply:
This is a temporary commit to test the changes to check if they break any existing tests.

I will clean up once its done.